File: example-projects/junit5-report-portal/src/test/java/com/tngtech/jgiven/example/projects/junit5/ThenStage.java
package com.tngtech.jgiven.example.projects.junit5;

import com.tngtech.jgiven.Stage;
import com.tngtech.jgiven.annotation.ScenarioState;
import org.junit.jupiter.api.Assertions;

public class ThenStage extends Stage<ThenStage> {
    @ScenarioState(required = true)
    String result;

    public void the_result_is(String expectedResult) {
        Assertions.assertEquals(expectedResult, result);
    }
}

File: example-projects/kotlin/src/test/kotlin/com/tngtech/jgiven/example/projects/junit5/ThenStage.kt
package com.tngtech.jgiven.example.projects.junit5

import com.tngtech.jgiven.Stage
import com.tngtech.jgiven.annotation.ScenarioState
import org.junit.jupiter.api.Assertions

open class ThenStage : Stage<ThenStage>() {

    @ScenarioState(required = true)
    lateinit var result: String

    open fun the_result_is(expectedResult: String) {
        Assertions.assertEquals(expectedResult, result)
    }
}
